src/twobinomial.jl

```diff
@@ -0,0 +1,541 @@
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export TwoSampleBinomialTest
+
+struct TwoSampleBinomialTask
+    x1::Int
+    n1::Int
+    x2::Int
+    n2::Int
+    δ::Float64
+    ptype::Symbol
+    htype::Symbol
+    alpha::Float64
+    method::Symbol
+end
+
+struct TwoSampleBinomialTest <: HypothesisTest
+    test::TwoSampleBinomialTask
+    est
+    se
+    ci
+    result
+end
+
+function cihalpha(htype, alpha)
+    if  htype == :equivalence || htype == :superiority
+        cialpha = alpha * 2
+    elseif htype == :equality
+        cialpha = alpha
+    else
+        throw(ArgumentError("unknown hypothesis=$(htype)"))
+    end
+    cialpha
+end
+
+"""
+    TwoSampleBinomialTest(x1::Real, n1::Real, x2::Real, n2::Real, δ::Real; ptype::Symbol, htype::Symbol, alpha::Real = 0.05, method::Symbol = :default)
+
+Perform a two sample binomial hypothesis check via computing confidence intrval
+with coverage `1 - alpha` for `equality` test or `1 - 2 * alpha` for `equivalence`
+or `superiority` test.
+
+Hypothesis type `htype` are:
+- `:equality`
+- `:equivalence`
+- `:superiority`
+
+Parameters type `ptype` are:
+    - `:difference` : risk difference;
+    - `:riskratio` : risk ratio;
+    - `:oddratio` : odd ratio;
+
+For each parameters can be used  following methods.
+
+Possible values for `method` for `difference` are:
+
+  - `:mn`
+  - `:wald`
+  - `:waldcc`
+  - `:nhs`
+  - `:nhscc`
+  - `:ac`
+  - `:ha`
+  - `:mnmee`
+  - `:mover`
+
+Possible values for `method` for `riskratio` are:
+
+  - `:mn`
+  - `:cli`
+  - `:li`
+  - `:mover`
+
+Possible values for `method` for `oddratio` are:
+
+  - `:mn`
+  - `:woolf`
+  - `:awoolf`
+  - `:mover`
+
+"""
+function TwoSampleBinomialTest(x1::Real, n1::Real, x2::Real, n2::Real, δ::Real; ptype::Symbol, htype::Symbol, alpha::Real = 0.05, method::Symbol = :default)
+    test = TwoSampleBinomialTask(x1, n1, x2, n2, δ, ptype, htype, alpha, method)
+    cialpha = cihalpha(htype, alpha)
+    if ptype == :difference
+        est, se, lci, uci = ci_diff(test; level = 1 - cialpha, method = method)
+    elseif ptype == :oddratio
+        est, se, lci, uci = ci_or(test; level = 1 - cialpha, method = method)
+    elseif ptype == :riskratio
+        est, se, lci, uci = ci_rr(test; level = 1 - cialpha, method = method)
+    else
+        throw(ArgumentError("unknown parameter=$(ptype)"))
+    end
+    result = false
+    if htype == :equivalence
+        if lci > -abs(δ) && uci < abs(δ) result = true end
+    elseif htype == :equality
+        if lci > δ || uci < δ result = true end
+    elseif htype == :superiority
+        if lci > δ result = true end
+    end
+    TwoSampleBinomialTest(test, est, se, (lci, uci), result)
+end
+################################################################################
```

Comment:
Just use empty lines to separate functions. Please also add comments explaining what each function does (i.e. to make it explicit what each kind of interval is). See what binomial.jl or other files look like.